update doc with fast + ioutils (#4)
ml-explore/mlx-c: docs/src/overview.rst

IO Operations
-------------

> MLX C wraps a number of :doc:`array IO operations <io>`, which saves and
> loads arrays in several common formats. See :doc:`IO utils <ioutils>` for

Reviewer: ```suggestion
MLX C wraps a number of :doc:`array IO operations <io>`, which save and
load arrays in several common formats. See :doc:`IO utils <ioutils>` for
```

fusing kernels, and reducing overhead. :doc:`Compilation operations <compile>`
are function transformations which take a closure and return a new closure
(which is the compiled version of the given closure).

Fast Custom Ops
===============

> To maximize performance, one can leverage :doc:`compilation
> <compile>`. However, sometimes compilation falls short, and in that case
> MLX comes with :doc:`fast <fast>` custom implementations for some
> mainstream operations.

Reviewer: ```suggestion
To maximize performance MLX has :doc:`fast <fast>` custom implementations for some
common operations.
```

(The stuff about compilation doesn't add much here IMO.)
